pyproject.toml

```diff
@@ -38,25 +38,30 @@ dependencies = [
 
 [project.optional-dependencies]
 tests = [
-    "pytest==7.4.3",
+    "pytest==8.0.2",
     "pytest-cov==4.1.0",
+    "flask",
 ]
 doc = [
-    "ansys-sphinx-theme==0.13.1",
+    "pyaedt[all]",
+    "PySide6-Essentials",
+    "ansys-sphinx-theme==0.14.0",
     "numpydoc==1.6.0",
     "recommonmark==0.7.1",
     "sphinx==7.2.6",
     "sphinx-copybutton==0.5.2",
     "enum-tools[sphinx]==0.11.0",
     "sphinx-autoapi==3.0.0",
-    "sphinx-autobuild==2021.3.14",
-    "sphinx-autodoc-typehints==1.25.2",
+    "sphinx-autobuild==2024.2.4",
+    "sphinx-autodoc-typehints==2.0.0",
     "sphinx-notfound-page==1.0.0",
     "nbsphinx==0.9.3",
     "sphinx_design",
     "jupytext",
-    "ipython==8.16.1",
+    "ipython==8.22.1",
     "jupyterlab==4.0.6",
+    "pypandoc",
+    "vtk==9.3.0",
```

Comment:
Please be sure that this doesn't break anything in your examples.
That was the case previously in pyaedt !